File: Caesar_Cipher/caesar_cipher.py
ALPHABET = ' ABCDEFGHIJKLMNOPQRSTUVWXYZ'

def encrypt(plain_text, key):
    cipher_text = ''
    for c in plain_text.upper():
        index = ALPHABET.find(c);
        index = (index + key) % len(ALPHABET)
        cipher_text = cipher_text + ALPHABET[index]
    return cipher_text

def decrypt(cipher_text, key):
    plain_text = ''
    for c in cipher_text:
        index = ALPHABET.find(c)
        index = (index - key) % len(ALPHABET)
        plain_text = plain_text + ALPHABET[index]
    return plain_text

if __name__ == '__main__':
    cipher_text = encrypt('I am COOL', 3)
    print(decrypt(cipher_text, 3))
    

File: Caesar_Cipher/CaesarCipher.java
package Caesar_Cipher;

public class CaesarCipher{
    private static final String ALPHABET = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static String encrypt(String freeText, int key){
        StringBuilder cipherText = new StringBuilder();
        for(char ch: freeText.toUpperCase().toCharArray()){
            int index = Math.floorMod(ALPHABET.indexOf(ch)+key, ALPHABET.length());
            cipherText.append(ALPHABET.charAt(index));
        }
        return cipherText.toString();
    }

    public static String decrypt(String cipherText, int key){
        StringBuilder freeText = new StringBuilder();
        for(char ch: cipherText.toCharArray()){
            int index = Math.floorMod(ALPHABET.indexOf(ch)-key, ALPHABET.length());
            freeText.append(ALPHABET.charAt(index));
        }
        return freeText.toString();
    }
    public static void main(String[] args) {
        String cipherText = encrypt("I am COOL", 3);
        System.out.println(decrypt(cipherText, 3));
    }
}